src/twisted/internet/defer.py

```diff
@@ -161,6 +172,151 @@ def maybeDeferred(f, *args, **kw):
 
 
 
+def extractCurrentResult(deferred, raises=True):
+    """
+    Extract the current result from a L{Deferred} that has already fired all of
+    its callbacks.
+
+    Take a deferred that has already fired all of its callbacks and extract and
+    return the result of the L{Deferred}. This process of extracting the result
+    will consume the result, error or otherwise. If the L{Deferred} has an
+    error then the process of extracting the result will, by default, re-raise
+    the original exception unless the L{raises} parameter is set to L{False}.
+
+    The primary purpose of this function is to allow code to be written that is
+    internally asynchronous but which can be consumed without an event loop in
+    a synchronous fashion by using L{extractCurrentResult} at the boundaries.
+    This enables libraries to be written that can function both with Twisted
+    and with synchronous code instead of requiring an asynchronous and a
+    synchronous implementation for all functionality.
+
+    @attention: This function CANNOT take a Deferred with pending asynchronous
+        work to be done and make it synchronous. It can ONLY extract results
+        from a L{Deferred} which has already fired its callbacks.
+
+    @type deferred: Deferred
+    @param deferred: The Deferred from which results will be extracted..
+
+    @type raises: bool
```

Comment:
This boolean flag still makes me uneasy personally but if others are ok with it maybe at least calling it `reraise` reads a bit more straightforwardly? `extractCurrentResult(raises=True)` vs. `extractCurrentResult(reraise=True)`